plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogConfig.java

```diff
@@ -15,16 +15,34 @@
 
 import io.airlift.configuration.Config;
 import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.ConfigSecuritySensitive;
+import io.airlift.units.Duration;
+import io.airlift.units.MinDuration;
+import org.projectnessie.client.NessieConfigConstants;
+
 import jakarta.validation.constraints.NotEmpty;
 import jakarta.validation.constraints.NotNull;
 
 import java.net.URI;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
 
 public class IcebergNessieCatalogConfig
 {
+    public enum Security
+    {
+        NONE,
+        BEARER
+    }
+
     private String defaultReferenceName = "main";
     private String defaultWarehouseDir;
     private URI serverUri;
+    private Security security = Security.NONE;
+    private Optional<String> bearerToken = Optional.empty();
+    private Duration readTimeout = new Duration(NessieConfigConstants.DEFAULT_READ_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS);
```

Comment:
For the sake of an easier review from the maintainers please consider splitting the non security related settings to a new commit.

Reply:
done